test/geometry/calibration/test_pnp.py

```diff
@@ -0,0 +1,137 @@
+import pytest
+import torch
+from torch.autograd import gradcheck
+
+import kornia
+from kornia.testing import assert_close, tensor_to_gradcheck_var
+
+
+class TestSolvePnpDlt:
+
+    @staticmethod
+    def _get_samples(shape, low, high, device, dtype):
+        """Return a tensor having the given shape and whose values are in the range [low, high)"""
+        return ((high - low) * torch.rand(shape, device=device, dtype=dtype)) + low
+
+    @staticmethod
+    def _project_to_image(world_points, world_to_cam_4x4, repeated_intrinsics):
+        r"""Projects points in the world coordinate system to the image coordinate system.
+
+        Since cam_points will have shape (B, N, 3), repeated_intrinsics should have
+        shape (B, N, 3, 3) so that kornia.geometry.project_points can be used.
+        """
+        cam_points = kornia.geometry.transform_points(world_to_cam_4x4, world_points)
+        img_points = kornia.geometry.project_points(cam_points, repeated_intrinsics)
+
+        return img_points
+
+    def _create_world_to_cam(self, device, dtype):
+        """Creates a ground truth world to cam matrix that can be used for the test data."""
+        tau = 2 * 3.141592653589793
+        angle_axis = self._get_samples(
+            shape=(1, 3), low=-tau, high=tau, dtype=dtype, device=device,
+        )
+        translation = self._get_samples(
+            shape=(3,), low=-100, high=100, dtype=dtype, device=device,
+        )
+
+        rotation = kornia.geometry.angle_axis_to_rotation_matrix(angle_axis)
```

Comment:
I think the best here is to create a random quaternion and convert to rotation_matrix to assure that is orthogonal and has determinant == +1
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.special_ortho_group.html#scipy.stats.special_ortho_group